telegram/ext/conversationhandler.py

```diff
@@ -596,20 +602,25 @@ def handle_update(  # type: ignore[override]
             new_state = exception.state
             raise_dp_handler_stop = True
         with self._timeout_jobs_lock:
-            if self.conversation_timeout and dispatcher.job_queue:
-                # Add the new timeout job
-                if isinstance(new_state, Promise):
-                    new_state.add_done_callback(
-                        functools.partial(
-                            self._schedule_job,
-                            dispatcher=dispatcher,
-                            update=update,
-                            context=context,
-                            conversation_key=conversation_key,
+            if self.conversation_timeout:
+                if dispatcher.job_queue is not None:
+                    # Add the new timeout job
+                    if isinstance(new_state, Promise):
+                        new_state.add_done_callback(
+                            functools.partial(
+                                self._schedule_job,
+                                dispatcher=dispatcher,
+                                update=update,
+                                context=context,
+                                conversation_key=conversation_key,
+                            )
+                        )
+                    else:
+                        self._schedule_job(
+                            new_state, dispatcher, update, context, conversation_key
                         )
-                    )
                 else:
-                    self._schedule_job(new_state, dispatcher, update, context, conversation_key)
+                    self.logger.warning("`conversation_timeout` can't work without JobQueue!")
```

Comment:
```suggestion
                    self.logger.warning("Ignoring `conversation_timeout` because the Dispatcher has no JobQueue.")
```

Let's try to be precise both for the users and for future-us 😉